mapper-javabean/src/main/java/org/hibernate/search/v6poc/entity/javabean/model/impl/JavaBeanTypeModel.java

```diff
@@ -10,39 +10,69 @@
 import java.beans.IntrospectionException;
 import java.beans.Introspector;
 import java.beans.PropertyDescriptor;
+import java.lang.annotation.Annotation;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.Stream;
 
 import org.hibernate.search.v6poc.entity.pojo.model.spi.PropertyModel;
 import org.hibernate.search.v6poc.entity.pojo.model.spi.TypeModel;
 import org.hibernate.search.v6poc.util.SearchException;
 
-public class JavaBeanTypeModel<T> implements TypeModel<T> {
+class JavaBeanTypeModel<T> implements TypeModel<T> {
 
+	private final JavaBeanIntrospector introspector;
 	private final Class<T> type;
 	private final BeanInfo beanInfo;
+	private final BeanInfo declaredBeanInfo;
 
-	public JavaBeanTypeModel(Class<T> type) throws IntrospectionException {
+	JavaBeanTypeModel(JavaBeanIntrospector introspector, Class<T> type) throws IntrospectionException {
+		this.introspector = introspector;
 		this.type = type;
 		this.beanInfo = Introspector.getBeanInfo( type );
+		this.declaredBeanInfo = Introspector.getBeanInfo( type, type.getSuperclass() );
 	}
 
 	@Override
 	public Class<T> getJavaType() {
 		return type;
 	}
 
+	@Override
+	public <A extends Annotation> Optional<A> getAnnotationByType(Class<A> annotationType) {
+		return introspector.getAnnotationByType( type, annotationType );
+	}
+
+	@Override
+	public <A extends Annotation> Stream<A> getAnnotationsByType(Class<A> annotationType) {
+		return introspector.getAnnotationsByType( type, annotationType );
+	}
+
+	@Override
+	public Stream<? extends Annotation> getAnnotationsByMetaAnnotationType(Class<? extends Annotation> metaAnnotationType) {
+		return introspector.getAnnotationsByMetaAnnotationType( type, metaAnnotationType );
+	}
+
 	@Override
 	public PropertyModel<?> getProperty(String propertyName) {
 		try {
 			String normalizedName = Introspector.decapitalize( propertyName );
 			PropertyDescriptor propertyDescriptor = getPropertyDescriptor( normalizedName );
-			return new JavaBeanPropertyModel<>( this, propertyDescriptor.getPropertyType(), propertyDescriptor );
```

Comment:
OK, so, it's coming a bit out of the blue but seeing `JavaBeanPropertyModel` just made me think about it: we had several requests for HV of people wanting us to support different models (especially the generation framework generating `property()` as a getter for property).

See this one: https://hibernate.atlassian.net/browse/HV-1363 .

Reply:
Yeah, it was not about sharing code but more than it has been a common requirement these days so making it easy for the user to have another strategy (for common cases) might be interesting.

It was more a thought I wanted to share than anything else.